### PR TITLE
support non-gitlab hosted urls

### PR DESCRIPTION
**Pull request by kobik**
Fixing https://github.com/repo-utils/parse-github-repo-url/issues/5

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,8 +9,7 @@ module.exports = function (string) {
   // normalize git@ and https:git@ urls
   string = string.replace(/^git@/, 'https://')
   string = string.replace(/^https:git@/, 'https://')
-  string = string.replace('.com:', '.com/')
-
+  string = string.replace(/\.([A-Za-z]+)\:/, '.$1/');
```
Comment (hutson) [inline index.js]: The pattern used for matching the TLD will need to be changed at a later time. Non-ascii characters are allowed, and used, for international domains.

File: index.js [hunk 2]
```diff
@@ -38,7 +37,8 @@ module.exports = function (string) {
   if (m) return m.slice(1, 4)
 
   // https://docs.gitlab.com/ce/user/group/subgroups/
-  if (~url.host.indexOf('gitlab')) {
+  // https://git.company.com/ce/user/group/subgroups/
+  if (~url.host.indexOf('git')) {
```
Comment (hutson) [inline index.js]: Is it worth checking for `git` at this point? There are GitLab deployments out there that do not have `git` in their name.

What would happen if we did not check for `git` and just attempted to process the URL as is?

File: test.js [hunk 3]
```diff
@@ -86,6 +92,18 @@ describe('gitlab urls', function () {
...
   })
 
+  it('parses https non-itlab url', function () {
+    var url = 'https://git.mycompany.com/user/test1.git'
+    var parsed = parse(url)
+    assert.deepEqual(['user', 'test1', ''], parsed)
+  })
+
+  it('parses https non-itlab url', function () {
+    var url = 'https://git.mycompany.local/user/test1.git'
```
Comment (hutson) [inline test.js]: Can this either be combined with the above test case, or have the description updated to reflect that you're testing a `local` TLD of a GitLab deployment?

File: index.js (hunk 2 above)
Comment (kobik) [reply index.js]: I was debating with myself about this one as well. I left it, as I wasn't sure about the motivation behind this condition.

If we both thinks the same, I guess we should remove it.

File: index.js (hunk 1 above)
Comment (kobik) [reply index.js]: Good point

**Review by hutson — APPROVED** [review]

**Pull request closed without merging**